### PR TITLE
Add openwebstart 0.5.1

### PR DESCRIPTION
**Pull request by tiiiecherle**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you, but we’ll be rejecting this on account of [not being notable enough](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/rejected_casks.md):

> + The Cask is too obscure. Examples:
>   + A self-submitted app from a GitHub repository that is [not notable enough](https://github.com/Homebrew/homebrew-cask/pull/28103) (under 50 stars).
>   + [Electronic Identification (eID) software](https://github.com/Homebrew/homebrew-cask/issues/59021).

**Comment by tiiiecherle** [comment]
@vitorgalvao Thanks for the feedback. While the restructuring of all java/openjdk formulae and casks is going on oracle java is definitely out. With oracle java8 out java web start is out and there is  (of my knowledge) no formula or cask to bring back this functionality. But this one does. It will be 1.0.0 soon and will still be needed by all people that depend on jnlp installer.

It would be nice to reconsider this one and see how it goes.

Thanks

**Comment by barthel** [comment]
I know, this software is used as a group-wide strategy of a large corporation (see sponsor webpage) to roll out software.
This corporation supports the development and is, with its thousands of employees, one of the early adopters.

**Comment by vitorgalvao** [comment]
> It would be nice to reconsider this one and see how it goes.

And we’ll do so once it reaches the notability threshold, provided someone resubmits it. Adding it now because it might be popular in the future is premature.

> Thanks

Thank *you*. I know you’re a good contributor, and rest assured I value your input. However, in this case deferring inclusion makes more sense. I’ve seen casks in this situation reach the requirements in a day.

> I know, this software is used as a group-wide strategy of a large corporation

Which in no way indicates anyone at said corporation uses HBC.

**Comment by tiiiecherle** [comment]
> Thank _you_. I know you’re a good contributor, and rest assured I value your input. However, in this case deferring inclusion makes more sense. I’ve seen casks in this situation reach the requirements in a day.

@vitorgalvao Thanks for the positive feedback. In the end the homebrew users profit from the great work you and the homebrew team is doing. I take that chance to thank you all for that, very much appreciated ;)))

For the acceptance of the pull request I respect the decision and the underlying reasons. I just voted for the project and gave it one more star at least...

Have a nice evening ;)

**Comment by barthel** [comment]
> I’ve seen casks in this situation reach the requirements in a day.
The OpenWebStart project (almost) reached the 50 stars now. :-)

@tiiiecherle Would you reopen your pull request or make a new one?

@vitorgalvao Are there any other reasons against an integration other than the missing stars of the project?

Many thanks to both of you and for your time invested.

**Comment by tiiiecherle** [comment]
@barthel I can do that tonight after @vitorgalvao gives his ok ;)

**Comment by hendrikebbers** [comment]
The project already has > 50 stars now :)

File: Casks/openwebstart.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  sha256 '96285f6592cae0093af2acb9a4f5865338edad292dcdd4e565e34bc5485e3f40'
+
+  # github.com/karakun/OpenWebStart was verified as official when first introduced to the cask
+  url "https://github.com/karakun/OpenWebStart/releases/download/v#{version}/OpenWebStart_macos_#{version.dots_to_underscores}.dmg"
+  appcast 'https://github.com/karakun/OpenWebStart/releases.atom'
+  name 'OpenWebStart'
+  homepage 'https://openwebstart.com/'
+
+  installer script: {
+                      executable:   "#{staged_path}/OpenWebStart Installer.app/Contents/MacOS/JavaApplicationStub",
```
Comment (vitorgalvao) [inline Casks/openwebstart.rb]: Is it possible to install to a custom directory, and make that directory `appdir`?

Comment (barthel) [reply Casks/openwebstart.rb]: @hendrikebbers could that answer best, I think.

**Comment by barthel** [comment]
@vitorgalvao Thanks for restart the game.

File: Casks/openwebstart.rb (hunk 1 above)
Comment (hendrikebbers) [reply Casks/openwebstart.rb]: You can install OWS to any directory you want. If you want to define that dir for the headless installer you need to create a config file and pass it to the installer. Is that possible?

**Comment by barthel** [comment]
@tiiiecherle Please contact me if I can support you.

@hendrikebbers Could you please provide a small example of the config file and how this file will be passed to the installer?

**Comment by tiiiecherle** [comment]
> @tiiiecherle Please contact me if I can support you.

@barthel Thanks. But I´m sorry I lost you somewhere on the way ;) Support for which project?
If you mean this cask I already did a small improvement regarding the uninstaller to be run unattended. For what I can see it is good to go ;)

File: Casks/openwebstart.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/openwebstart.rb]: > If you want to define that dir for the headless installer you need to create a config file and pass it to the installer. Is that possible?

Yes, but more work than it’s worth.

**Pull request merged**